ftag/hdf5/h5reader.py

```diff
@@ -240,7 +256,15 @@ def load(
         return {name: np.concatenate(array) for name, array in data.items()}
 
     def estimate_available_jets(self, cuts: Cuts, num: int = 1_000_000) -> int:
-        """Estimate the number of jets available after selection cuts, rounded down."""
+        """Estimate the number of jets available after selection cuts, rounded down.
+
+        If self.equal_jets is True, the number of jets is estimated from the smallest file.
+        However, the remaining jets after cuts is estimated from all files.
+        """
         all_jets = self.load({self.jets_name: cuts.variables}, num)[self.jets_name]
-        estimated_num_jets = len(cuts(all_jets).values) / len(all_jets) * self.num_jets
+        est_total_jets = {
+            True: min(r.num_jets for r in self.readers) * len(self.readers),
+            False: self.num_jets,
+        }[self.equal_jets]
```

Comment:
Thanks @AndriusVaitkus97. Is there a reason to store both options in a dict rather than using a simple if/else to assign a single variable?

I'm also not completely comfortable with the assumption that the different samples would pass the cuts with the same frequency. Probably the most robust option is to implement a load method for the single readers and properly estimate the number of jets available in each sample independently. 

We could also have a couple of separate tests for the `estimate_available_jets` function. 

Happy to implement these as follow-up to this MR if you prefer.

Reply:
Hey @samvanstroud,

> Is there a reason to store both options in a dict rather than using a simple if/else to assign a single variable? 

Not really, I can replace that with an if/else instead.

> I'm also not completely comfortable with the assumption that the different samples would pass the cuts with the same frequency.

Fair enough, but do you think if everything else works, we should merge this and create a separate pull request for improving `estimate_available_jets`? The functionality of `equal_jets` flag is done and tested. Same goes for the test of `estimate_available_jets`.

Reply:
Ok sounds good, if you can move the dict to an if/else (it just looks weird to me) and undraft then we can merge this and follow up in another MR